### PR TITLE
Switch FxCop to PrivateAssets="All"

### PR DESCRIPTION
**Pull request by danielmarbach**
## Proposed Changes

Outcome of https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/932#discussion_r488823971

By switching FxCop analyzer to private assets all the analyzer exclusion can be removed.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [x] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

### CODE REVIEW

File: projects/RabbitMQ.Client/RabbitMQ.Client.csproj [hunk 1]
```diff
@@ -57,7 +57,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0" PrivateAssets="None" />
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0" PrivateAssets="All" />
```
Comment (adamralph) [inline projects/RabbitMQ.Client/RabbitMQ.Client.csproj:60]: 👍 it's very important to make this change.

As it is currently, the `RabbitMQ.Client` package has a _runtime_ dependency on the `Microsoft.CodeAnalysis.FxCopAnalyzers` package.

**Review by adamralph — APPROVED** [review]

File: projects/RabbitMQ.Client/RabbitMQ.Client.csproj (hunk 1 above)
Comment (danielmarbach) [reply projects/RabbitMQ.Client/RabbitMQ.Client.csproj:60]: The previous PR was only merged to master. No kittens have been harmed yet :)

**Review by lukebakken — APPROVED** [review]
Thanks everyone for noticing!

**Comment by danielmarbach** [comment]
Appveyor is stuck

**Pull request merged**